### PR TITLE
Pkg controller test requests 2

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/spec/controllers/webui/package_controller_spec.rb [hunk 1]
```diff
@@ -2,20 +2,89 @@
...
 RSpec.describe Webui::PackageController, vcr: true do
   let(:admin) { create(:admin_user, login: "admin") }
-  let(:user) { create(:user, login: 'tom') }
+  let(:user) { create(:confirmed_user, login: 'tom') }
   let(:source_project) { user.home_project }
   let(:source_package) { create(:package, name: 'my_package', project: source_project) }
   let(:target_project) { create(:project) }
+  let(:package) { create(:package_with_file, name: "package_with_file", project: source_project) }
 
   describe 'submit_request' do
```
Comment (mdeniz) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: Is better to change the describe desctiption to "POST #submit_request" ?

File: src/api/spec/controllers/webui/package_controller_spec.rb [hunk 2]
```diff
@@ -2,20 +2,89 @@
...
+
+      it { expect(flash[:notice]).to match("Created .+submit request \\d.+to .+#{target_project}") }
+      it { expect(response).to redirect_to(package_show_path(project: source_project, package: package)) }
+
+      it "creates a submit request" do
+        expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: package.name)).to exist
+      end
+    end
+
+    context "having whitespaces in parameters" do
```
Comment (mdeniz) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: Would be better to have a shared example called "submit request succesful created" and set the params in each case from outside?

File: src/api/spec/controllers/webui/package_controller_spec.rb [hunk 3]
```diff
@@ -2,20 +2,89 @@
...
+
     context 'not successful' do
       before do
         login(user)
         post :submit_request, { project: source_project, package: source_package, targetproject: target_project.name }
       end
 
       it { expect(flash[:error]).to eq('Unable to submit: The source of package home:tom/my_package is broken') }
-      it { expect(BsRequestActionSubmit.where(target_project: target_project, target_package: source_package).count).to eq(0) }
+      it { expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: source_package.name)).not_to exist }
```
Comment (mdeniz) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: Can you add the `redirect_to` spec like in line 59  of this file?

File: src/api/spec/controllers/webui/package_controller_spec.rb [hunk 4]
```diff
@@ -2,20 +2,89 @@
...
+        expect(BsRequestActionSubmit.where(target_project: source_project.name, target_package: package.name)).not_to exist
+      end
+    end
+
+    context "invalid request (missing parameters)" do
+      before do
+        login(user)
+        post :submit_request, { project: source_project, package: "", targetproject: source_project }
+      end
+
```
Comment (mdeniz) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: Can you add the same expectation as line 76, or this creates the submit request?

File: src/api/spec/controllers/webui/package_controller_spec.rb (hunk 2 above)
Comment (bgeuken) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: That's a good idea. Let me try.

File: src/api/spec/controllers/webui/package_controller_spec.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: It was already like that:/ I changed it now. Though this required to update the VCR cassettes. That's why I put it into a new commit

**Comment by bgeuken** [comment]
@mdeniz Thanks for the input. I created a new PR and will close this one now.

**Pull request closed without merging**